app/src/forms/proxy/service.js

```diff
@@ -19,15 +19,38 @@ const trimLeadingSlashes = (str) => str.replace(/^\/+|\$/g, '');
 const trimTrailingSlashes = (str) => str.replace(/\/+$/g, '');
 
 const service = {
+  _getIds: async (payload) => {
+    let formId = payload['formId'];
+    let versionId = payload['versionId'];
+    let submissionId = payload['submissionId'];
+    // when we are provided with a submission id (ex. submitting a draft submission)
+    // we need to fetch the related form and version id (if not provided)
+    if (submissionId) {
+      const meta = await SubmissionMetadata.query().where('submissionId', submissionId).first();
+      if (meta) {
+        formId = meta.submissionId;
+        versionId = meta.formVersionId;
+        submissionId = meta.submissionId;
+      }
+    }
```

Comment:
The code coverage report is complaining about these lines:

![image](https://github.com/user-attachments/assets/8e082c5a-b120-485b-855a-6206a97103d2)

Reply:
how do i see the code coverage report?

Reply:
I'll create a Task to add that to the docs!
- locally for the backend it's in `/app/coverage/lcov-report/index.html`
- for a PR it's in the artifacts for the "Tests" Action:
![image](https://github.com/user-attachments/assets/9cd6cc3c-783a-4bfc-a36e-35fedae20001)

Reply:
probably on the long list of backlog items but maybe the metrics should somehow be in your face? like easily seeing that this PR reduced coverage by X% or something?

Reply:
That would be a great feature! A lot of stuff happens after the merge, etc, due to permissions. But yes, it's something that would be great if someone looked into it.